client/src/app/admin-dashboard/admin-dashboard.component.ts

```diff
@@ -18,11 +34,15 @@ export class AdminDashboardComponent implements OnInit {
   adminForm: FormGroup;
   adminEuriborDate!: string;
 
-  validatorsNum = [Validators.required, Validators.pattern(/^\d+\.?\d*$/)];
+  validatorsNum = [
+    Validators.required,
+    Validators.pattern(/^\d+\.?\d*$/),
+    checkIfLessThanZero,
+  ];
   validatorsPercent = [
     ...this.validatorsNum,
-    Validators.min(0),
     Validators.max(100),
+    checkIfLessThanZero,
```

Comment:
You don't need to provide it twice. It's already in `validatorsNum`

Reply:
oh, yes....